### DISCUSSION TITLE
make-stmconst.py script failed to parse STM32CubeU5 header file

### DISCUSSION THREAD

**Discussion by jandriea** [micropython/micropython#9299] [Core Development] [answered]
Hello all,

Currently, I am trying to port the STM32U5 family to Python. 

I have updated the lib/stm32lib repository to include the STM32CubeU5 that I got from here
https://github.com/STMicroelectronics/STM32CubeU5.git
However, I always failed to compile the code because there is a struct that the make-stmconst.py script failed to parse.

this code snippet is taken from https://github.com/STMicroelectronics/STM32CubeU5/blob/main/Drivers/CMSIS/Device/ST/STM32U5xx/Include/stm32u575xx.h at lines 2063-2070

```c
/**
  * @brief RSSLib secure callable function pointer structure
  */
typedef struct
{
  __IM uint32_t Reserved2[2];
  __IM RSSLIB_S_CloseExitHDP_TypeDef CloseExitHDP;        /*!< RSSLIB Bootloader Close and exit HDP  Address offset: 0x28 */
}S_pFuncTypeDef;
```

After some debugging, I found that the script failed to parse the line because of this particular regex at lines 73-76
```py
(
    "} TypeDef",
    re.compile(r"} *(?P<id>[A-Z][A-Za-z0-9_]+)_(?P<global>([A-Za-z0-9_]+)?)TypeDef;$"),
),
```
AFAIK, this regex will only match the line if the group **id** is consist of two characters or more. So the script will always fail to parse the struct with a single char in the group **id**.

Can we change the regex to make the script able to parse the struct?
something like
```py
(
    "} TypeDef",
    re.compile(r"} *(?P<id>[A-Za-z0-9_]+)_(?P<global>([A-Za-z0-9_]+)?)TypeDef;$"),
),
```

Thank you for your help

**Comment by dpgeorge** [comment, marked as answer]
Yes we can change that regex.  Although maybe it's a little more precise to have `(?P<id>[A-Z][A-Za-z0-9_]*)`

**Reply by jandriea** [reply]
Thanks for your reply, I will create a pull request for this update

**Reply by dpgeorge** [reply]
Fixed by 30e50ab1951680a046c224c12f7cc29aed410e2a
